resource/event_log.go

```diff
@@ -0,0 +1,840 @@
+	0x80000004: "EV_EFI_BOOT_SERVICES_DRIVER",
+	0x80000005: "EV_EFI_RUNTIME_SERVICES_DRIVER",
+	0x80000006: "EV_EFI_GPT_EVENT",
+	0x80000007: "EV_EFI_ACTION",
+	0x80000008: "EV_EFI_PLATFORM_FIRMWARE_BLOB",
+	0x80000009: "EV_EFI_HANDOFF_TABLES",
+	0x8000000A: "EV_EFI_PLATFORM_FIRMWARE_BLOB2",
+	0x8000000B: "EV_EFI_HANDOFF_TABLES2",
+	0x8000000C: "EV_EFI_VARIABLE_BOOT2",
+	0x80000010: "EV_EFI_HCRTM_EVENT",
+	0x800000E0: "EV_EFI_VARIABLE_AUTHORITY",
+	0x800000E1: "EV_EFI_SPDM_FIRMWARE_BLOB",
+	0x800000E2: "EV_EFI_SPDM_FIRMWARE_CONFIG",
+	0x401:      "PCR_MAPPING",
+	0x402:      "HASH_START",
+	0x403:      "COMBINED_HASH",
+	0x404:      "MLE_HASH",
+	0x40a:      "BIOSAC_REG_DATA",
+	0x40b:      "CPU_SCRTM_STAT",
+	0x40c:      "LCP_CONTROL_HASH",
+	0x40d:      "ELEMENTS_HASH",
+	0x40e:      "STM_HASH",
+	0x40f:      "OSSINITDATA_CAP_HASH",
+	0x410:      "SINIT_PUBKEY_HASH",
+	0x411:      "LCP_HASH",
+	0x412:      "LCP_DETAILS_HASH",
+	0x413:      "LCP_AUTHORITIES_HASH",
+	0x414:      "NV_INFO_HASH",
+	0x416:      "EVTYPE_KM_HASH",
+	0x417:      "EVTYPE_BPM_HASH",
+	0x418:      "EVTYPE_KM_INFO_HASH",
+	0x419:      "EVTYPE_BPM_INFO_HASH",
+	0x41a:      "EVTYPE_BOOT_POL_HASH",
+	0x4ff:      "CAP_VALUE",
+}
+
+var eventLogInfo *EventLogInfo
+
+//InitializeEventLogInfo - Function to create and intialize eventLogInfo variable
+func InitializeEventLogInfo() *EventLogInfo {
+	log.Trace("resource/event_log:InitializeEventLogInfo() Entering")
+	defer log.Trace("resource/event_log:InitializeEventLogInfo() Leaving")
+	eventLogInfo := new(EventLogInfo)
+	eventLogInfo.TxtHeapBaseAddr = 0
+	eventLogInfo.TxtHeapSize = 0
+	eventLogInfo.UefiEventAddr = 0
+	eventLogInfo.UefiEventSize = 0
+	eventLogInfo.TxtEnabled = false
+	return eventLogInfo
+}
+
+//GetEventLogInfo - Getter function for eventLogInfo variable
+func GetEventLogInfo() *EventLogInfo {
+	log.Trace("resource/event_log:GetEventLogInfo() Entering")
+	defer log.Trace("resource/event_log:GetEventLogInfo() Leaving")
+	if eventLogInfo == nil {
+		eventLogInfo = InitializeEventLogInfo()
+	}
+	return eventLogInfo
+}
+
+//FetchUefiEventInfo - Function to get Uefi Events Info
+func (eventLogInfo *EventLogInfo) FetchUefiEventInfo() error {
+
+	log.Trace("resource/event_log:FetchUefiEventInfo() Entering")
+	defer log.Trace("resource/event_log:FetchUefiEventInfo() Leaving")
+
+	uefiBaseAddress := make([]byte, Uint64Size)
+	uefiSize := make([]byte, Uint32Size)
+	file, err := os.Open(Tpm2FilePath)
+	if err != nil {
+		return errors.Wrapf(err, "resource/event_log:FetchUefiEventInfo() There was an error opening %s", Tpm2FilePath)
+	}
+	defer file.Close()
+
+	fileInfo, err := file.Stat()
+	if err != nil {
+		return errors.Wrapf(err, "resource/event_log:FetchUefiEventInfo() There was an error getting FileInfo of %s", Tpm2FilePath)
+	}
+
+	// As TPM2 file contains uefi_event_log size from offset 64-67 and uefi_event_log address from offset 68-75, so below check is for uefi_event_log Info in TPM2 file
+	if fileInfo.Size() < 76 {
```

Comment:
Check the binary's 'magic' make sure that is 'TPM2' (to make sure we're not parsing garbage).